app/controllers/api/graphql_controller.rb

```diff
@@ -28,6 +28,20 @@ def api_request?
       true
     end
 
+    def authenticate
+      if bruteforce_attempt?
+        log_bruteforce
+        render_error('Bruteforce attempt.', status: :unauthorized)
+        return false
+      end
+
+      authenticated = super
+
+      count_login_failure if available_sso.present? && !authenticated
```

Comment:
do we actually need the `available_sso.present?` condition here? might be good to also block attempts that don't pass the correct headers (bearer in the case of graphql)

Reply:
Yeah, we need this as a user can access the API without being signed in, e.g. to sign in.

Reply:
shouldn't we then `skip_before_action :authenticate` on the login action?

Reply:
It's not a dedicated action, just a mutation in the graph. So we cannot do that.